includes/Structs.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef STRUCTS_HPP
+# define STRUCTS_HPP
+
+#include <vector>
+#include <map>
+
+enum RequestTypes {
+    GET,
+    DELETE,
+    POST,
+};
+
+enum SectionTypes {
+    HTTP,
+    SERVER,
+    LOCATION,
+    UNKNOWN,
+};
+
+struct LocationConfig {
+    std::vector<RequestTypes> allowedRequestTypes;
+    std::string locationPath;
+};
+
+struct ServerConfig {
+    std::string indexFile;
+    std::string serverName;
+    int clientMaxBodySize;
```

Comment:
i would suggest using unsigned ints instead of ints or creating an int checker function with input of MIN and MAX values that will throw if not within or wrong type, not numbers etc. for exmaple as template or just function with overloads 
void limit(int value, int min, int max) const; <- for int check with overload 
void limit(std::string, int min, int max) const; <- for string lenght 
and so on, we can add different overloads depending on whats needed but it could add to simplicity of using because then when we introduce some new value we just limit(..) it. Or as a Template it can even act as an ultimate parser that will return whatever we needed by using stream. Just and idea xD

Reply:
Not entirely sure what you mean. I did make a helper function that checks for int max and negative values.


See `utils.cpp`

Reply:
I just had this idea of kinda unifying parsing instead of always writing checks. But this works too!

Reply:
Still not entirely sure what you mean/referring to :/